### DISCUSSION TITLE
Can't Connect Ollama With Lobe-Chat

### DISCUSSION THREAD

**Discussion by boudywho** [lobehub/lobehub#1486] [Self Hosting] [answered]
I will start by saying, I am no professional :)

I'm having trouble linking Lobe-Chat and Ollama in my Docker setup. Both containers are running (Ollama responds to curl), but I cannot connect them through the Lobe-Chat settings.

I run them both by ```sudo docker start lobe``` ```sudo docker start ollama```

Error:
```json
{
  "error": {
    "cause": {
      "errno": -111,
      "code": "ECONNREFUSED",
      "syscall": "connect",
      "address": "127.0.0.1",
      "port": 11434
    }
  },
  "endpoint": "http://127.0.***.1:****/",
  "provider": "ollama"
}
```

**Comment by boudywho** [comment]
I even tried running ```sudo docker run -p 11434:11434 ollama/ollama``` then running ```sudo docker run -d -p 3210:3210 -e OLLAMA_PROXY_URL=https://127.0.0.1:11434/ lobehub/lobe-chat```

Still didn't work, am I missing something here?

**Reply by arvinxx** [reply]
Plz check docs https://chat-docs.lobehub.com/en/self-hosting/examples/ollama

**Reply by boudywho** [reply]
I forgot to mention that I did run this docker command after installing Ollama locally ```docker run -d -p 3210:3210 -e OLLAMA_PROXY_URL=http://host.docker.internal:11434/v1 lobehub/lobe-chat``` this still didn't work

![image](https://github.com/lobehub/lobe-chat/assets/113399517/f5972fb0-edc4-4439-bf1a-07511cfcc00e)

**Reply by arvinxx** [reply]
just remove this part
<img width="675" alt="image" src="https://github.com/lobehub/lobe-chat/assets/28616219/26c22c2d-06b5-4f2f-8acd-b254964dd78b">

**Reply by boudywho** [reply]
![image](https://github.com/lobehub/lobe-chat/assets/113399517/b8c1a57a-8f35-4d69-b881-053703e9dd40)
I did try it, didn't work either, this time it gave "ENOTFOUND"

**Reply by arvinxx** [reply]
O I know, you just run ollama with docker. you should set the Proxy URL to your ollama docker container name, maybe like `http://ollama/v1` ?

**Reply by boudywho** [reply]
Unfortunately, this didn't work
![image](https://github.com/lobehub/lobe-chat/assets/113399517/ccc6a212-1d4f-49f2-ba43-7bc825e66df4)

**Reply by boudywho** [reply]
I even tried installing ollama locally (not in docker) and tried connecting it with Lobe, still no luck

**Reply by boudywho** [reply]
Vistting ```http://127.0.1:11434``` shows ```Ollama is running``` and when I do ```curl http://127.0.1:11434``` it shows also ```Ollama is running```

**Reply by boudywho** [reply]
![image](https://github.com/lobehub/lobe-chat/assets/113399517/29770694-267e-466d-8f27-255aa85b7ae5)

**Comment by ahag93** [comment]
++, I also tried this with the last docker image on this date (v0.133.1):
```
(base) ➜  _WORKSPACE_ curl http://127.0.0.1:11434           
Ollama is running%                                                                                                                       
(base) ➜  _WORKSPACE_ docker rm -f 92dc68ae6278
92dc68ae6278
(base) ➜  _WORKSPACE_ docker run --rm -d  --network=host -e OLLAMA_PROXY_URL=http://127.0.0.1:11434/ lobehub/lobe-chat:v0.133.1
1d8dd366962cd68dae7b41a4e53138699701888129d1e389a67e0ddcf87d88de
(base) ➜  _WORKSPACE_ docker logs -f 1d8dd366962cd68dae7b41a4e53138699701888129d1e389a67e0ddcf87d88de                          
   ▲ Next.js 14.1.3
   - Local:        http://localhost:3210
   - Network:      http://0.0.0.0:3210

 ✓ Ready in 71ms
Route: [ollama] OllamaBizError: {
  headers: {
  content-length: '18',
  content-type: 'text/plain',
  date: 'Fri, 08 Mar 2024 09:25:46 GMT'
},
  stack: 'Error: 404 404 page not found\n    at X.generate (/app/.next/server/edge-chunks/686.js:4:1745)\n    at tF.makeStatusError (/app/.next/server/edge-chunks/686.js:4:13387)\n    at tF.makeRequest (/app/.next/server/edge-chunks/686.js:4:14306)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async W.chat (/app/.next/server/edge-chunks/173.js:1:10559)\n    at async ei (/app/.next/server/edge-chunks/173.js:1:17810)\n    at async /app/.next/server/edge-chunks/212.js:6:64195\n    at async O.execute (/app/.next/server/edge-chunks/212.js:6:61088)\n    at async O.handle (/app/.next/server/edge-chunks/212.js:6:65462)\n    at async ey.handler (/app/.next/server/edge-chunks/212.js:7:31644)',
  status: 404
}
```

**Reply by arvinxx** [reply]
try with:
```
OLLAMA_PROXY_URL=http://0.0.0.0:11434/v1
```

**Reply by ahag93** [reply]
Thanks, but ollama doesn't expose an endpoint with **/v1**:

https://github.com/ollama/ollama/blob/main/docs/api.md
https://github.com/ollama/ollama/blob/main/api/client.go

**Reply by arvinxx** [reply]
please check: https://ollama.com/blog/openai-compatibility

**Reply by forge-oscar-caballero** [reply]
Hi, I faced this situation before, if you are using docker compose to run it please ensure to use this as your host url for Ollama:

```yml
version: '3.8'
 
services:
  lobe-chat:
    image: lobehub/lobe-chat
    container_name: lobe-chat
    restart: always
    ports:
      - '3210:3210'
    environment:
      OPENAI_API_KEY: sk-....
      OPENAI_PROXY_URL: https://api.openai.com/v1
      # ACCESS_CODE: lobe66
      OLLAMA_PROXY_URL: http://host.docker.internal:11434/v1

```


<img width="580" alt="Screenshot 2024-03-11 at 5 04 54 PM" src="https://github.com/lobehub/lobe-chat/assets/100241997/f56cf17b-009f-4760-8ea3-d481bc8ec82e">

Let me know if this works for you.

**Reply by ahag93** [reply]
Hi, Thank you for this information, indeed there is this doc Openai compatibility that I have not seen, after testing the compose, I have not had convincing results so far:

![2024-03-12 10 09 14](https://github.com/lobehub/lobe-chat/assets/12112148/f290ebd2-7376-4608-872e-099e9d55759b)

I don't know if you paid any attention to this note for the doc:
**OpenAI compatibility**

> Note: OpenAI compatibility is experimental and is subject to major adjustments including breaking changes. For fully-featured access to the Ollama API, see the Ollama [Python library](https://github.com/ollama/ollama-python), [JavaScript library](https://github.com/ollama/ollama-js) and [REST API](https://github.com/jmorganca/ollama/blob/main/docs/api.md).

thanks

**Comment by arvinxx** [comment, marked as answer]
I think it's fixed with `0.149.0`. If there is any other issue, please let me know

**Reply by walk4rever** [reply]
No, the problem is still there.

**Reply by arvinxx** [reply]
@walk4rever what's the problem?

**Reply by krenax** [reply]
@arvinxx I have the same error with v1.84.9.
